functests/vrf/vrf_sriov.go

```diff
@@ -0,0 +1,176 @@
+	sriovv1 "github.com/openshift/sriov-network-operator/api/v1"
+	sriovClean "github.com/openshift/sriov-network-operator/test/util/clean"
+	sriovtestclient "github.com/openshift/sriov-network-operator/test/util/client"
+	sriovcluster "github.com/openshift/sriov-network-operator/test/util/cluster"
+	sriovNamespaces "github.com/openshift/sriov-network-operator/test/util/namespaces"
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	resourceNameVrf = "sriovnicvrf"
+	testNetworkRed  = "test-vrf-sriov-network-red"
+	testNetworkBlue = "test-vrf-sriov-network-blue"
+)
+
+var _ = Describe("[sriov] VRF integration", func() {
+	describe := func(node string, ipStack string) string {
+
+		VRFParameters, err := newVRFTestParameters(node, ipStack)
+		if err != nil {
+			return fmt.Sprintf("error in parameters: node=%s, ipStack=%s", node, ipStack)
+		}
+		params, err := json.Marshal(VRFParameters)
+		if err != nil {
+			return fmt.Sprintf("error in parameters: node=%s, ipStack=%s", node, ipStack)
+		}
+
+		return fmt.Sprintf("%s", string(params))
+	}
+
+	apiclient := client.New("")
+	sriovclient := sriovtestclient.New("")
+	var nodesList []string
+
+	execute.BeforeAll(func() {
+		if discovery.Enabled() {
+			Skip("Discovery is not supported.")
+		}
+		err := sriovClean.All()
+		Expect(err).ToNot(HaveOccurred())
+		sriov.WaitStable(sriovclient)
+
+		err = namespaces.Create(sriovNamespaces.Test, apiclient)
+		Expect(err).ToNot(HaveOccurred())
+		sriovclient := sriovtestclient.New("")
+		sriovInfos, err := sriovcluster.DiscoverSriov(sriovclient, namespaces.SRIOVOperator)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(sriovInfos).ToNot(BeNil())
+		nodesList, err = nodes.MatchingOptionalSelectorByName(sriovInfos.Nodes)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(len(nodesList)).To(BeNumerically(">", 0))
+		sriovDevice, err := sriovInfos.FindOneSriovDevice(nodesList[0])
+		Expect(err).ToNot(HaveOccurred())
+		createSriovPolicy(sriovclient, sriovDevice, nodesList[0], 5, resourceNameVrf)
+
+		ipam := `{"type": "static"}`
+		metaPluginRed := fmt.Sprintf(`{"type": "vrf", "vrfname": "%s"}`, VRFRedName)
+		err = createSriovNetwork(sriovclient, sriovDevice, testNetworkRed, sriovNamespaces.Test, namespaces.SRIOVOperator, resourceNameVrf, ipam, metaPluginRed)
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(func() error {
+			netAttDef := &netattdefv1.NetworkAttachmentDefinition{}
+			return sriovclient.Get(context.Background(), runtimeclient.ObjectKey{Name: testNetworkRed, Namespace: sriovNamespaces.Test}, netAttDef)
+		}, 60*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+
+		metaPluginBlue := fmt.Sprintf(`{"type": "vrf", "vrfname": "%s"}`, VRFBlueName)
+		err = createSriovNetwork(sriovclient, sriovDevice, testNetworkBlue, sriovNamespaces.Test, namespaces.SRIOVOperator, resourceNameVrf, ipam, metaPluginBlue)
```

Comment:
You can leverage the CreateSriovNetwork from the sriov package. It takes an optional slice of functions for changing the network before creating it. This avoids the need for another function:

func CreateSriovNetwork(clientSet *testclient.ClientSet, intf *sriovv1.InterfaceExt, name string, namespace string, operatorNamespace string, resourceName string, ipam string, options ...SriovNetworkOptions) error {

options is a type SriovNetworkOptions func(*sriovv1.SriovNetwork)
so you can do:
sriovOptions := func(network *sriovv1.SriovNetwork) {
    network.MetaPlugins = blablabla
}

and pass it as the last argument to the above function